CONTRIBUTING.md

```diff
@@ -0,0 +1,481 @@
+
+Code block can highlight characters using the
+[syntax provided via `rehype-pretty-code`](https://rehype-pretty.pages.dev/#highlight-chars).
+
+In the meta string, place character segments between two `/` symbols. You can
+also highlight multiple segments of characters.
+
+Examples:
+
+````md
+```js /word/
+// this will be highlighted: word
+// this will partially highlight: hello-word
+```
+````
+
+````md
+```js /word/ /partially/
+// this will be highlighted: word
+// this will partially highlight: hello-word
+```
+````
+
+````md
+```rust filename="lib.rs" /word/ {1,3}
+// this will be highlighted: word
+// nothing highlighted
+// this will partially highlighted word and line: hello-word
+```
+````
+
+#### Diff lines
+
+Within a code block you can create a "diff" style line (i.e. a line was added or
+a line was removed) using the
+
+At the end of a line, add one of the following notations to the end of a line:
+
+- to show a red "line removed" diff line: `// [!code --]`
+- to show a green "line added" diff line: `// [!code ++]`
+
+Example:
+
+```ts
+const keep = "line that stays";
+const updated = "example that was should be removed"; // [!code --]
+const updated = "example that was should be added"; // [!code ++]
+const didNotChange = "this line did not change";
+```
+
+### blockquote
+
+Standard markdown quote blocks are rendered with our custom
+[Callout component](#callout).
+
+```md
+> This will be the Callout component with it's default styles
+```
+
+### Callout
+
+### Embed
+
+The `Embed` component can be used to handle custom rendering of specifically
+supported external content.
+
+Currently supported external content:
+
+- [YouTube videos](#youtube-video)
+- [Whimsical diagram](#whimsical-diagram)
+
+#### YouTube video
+
+To embed a YouTube video:
+
+```md
+<Embed url="https://youtu.be/eudSSvyZF9o" />
+```
+
+Or
+
+```md
+<Embed url="https://www.youtube.com/watch?v=eudSSvyZF9o" />
+```
+
+#### Whimsical diagram
+
+To embed a Whimsical diagram:
+
+```mdx
+<Embed url="https://whimsical.com/embed/PjBVAREV3DMyW5722tFKJ8" />
+```
+
+## Translations
+
+Content translations are supported via the Crowdin platform.
+
+All the markdown content committed to this repo is in the base language of
+`English`. On each successful merge of content, all changes are uploaded to
+Crowdin (via GitHub actions) for them to be translated into the supported
+locales.
+
+During production deployments of the developer content api, all the currently
+translated content is downloaded from Crowdin and deployed for solana.com to
+correctly render the user requested locale of content. Falling back to the base
+language when no translated content was found.
+
+## Local development
+
+This developer content repo is a NextJS application that serves the markdown
+based content as REST api for solana.com to consume. Think of it like a
+microservice for content.
+
+We call this microservice the "developer content api", or content api for short.
+
+### Setup locally
```

Comment:
what would be the reason a contributor would need to run this locally? since they won't have access to the solana-com frontend

maybe also mention that there is no frontend to display the content atm, it might be confusing when someone runs the repo (to check how their contribution looks) to find that its just a static page

Reply:
Would there a be a feasible way to run this API server, and the thing that consumes the API and shows the content, on localhost?